### PR TITLE
The new version of Regex (1.9.0) is not compatible with 1.8

### PR DESCRIPTION
**Pull request by sergey-shandar**
### Description

The new version of Regex (1.9.0) is not compatible with 1.8 if we use `regex::internal::*` See

https://github.com/Trust-Machines/stacks-sbtc/actions/runs/5466661304/jobs/9951913250

And the fix:

https://github.com/Trust-Machines/stacks-sbtc/pull/522

### Applicable issues
- fixes #

### Checklist
- [ ] Test coverage for new or modified code paths
- [ ] Changelog is updated
- [ ] Required documentation changes (e.g., `docs/rpc/openapi.yaml` and `rpc-endpoints.md` for v2 endpoints, `event-dispatcher.md` for new events)
- [ ] New clarity functions have corresponding PR in `clarity-benchmarking` repo
- [ ] New integration test(s) added to `bitcoin-tests.yml`

### CODE REVIEW

**Comment by kantai** [comment]
Thanks for finding this @sergey-shandar --

I'd suggest just deleting that line from `costs/mod.rs` instead:

```
diff --git a/clarity/src/vm/costs/mod.rs b/clarity/src/vm/costs/mod.rs
index ee9a1a6eb..c70ab9600 100644
--- a/clarity/src/vm/costs/mod.rs
+++ b/clarity/src/vm/costs/mod.rs
@@ -18,7 +18,6 @@ use std::collections::{BTreeMap, HashMap};
 use std::convert::{TryFrom, TryInto};
 use std::{cmp, fmt};
 
-use regex::internal::Exec;
 use rusqlite::types::{FromSql, FromSqlResult, ToSql, ToSqlOutput, ValueRef};
 use serde::{Deserialize, Serialize};
```

The `Exec` method isn't used anywhere, so we can just remove that import and then we don't need to version pin `regex`.

File: clarity/Cargo.toml [hunk 1]
```diff
@@ -23,7 +23,7 @@ rand_chacha = "=0.2.2"
 serde = "1"
 serde_derive = "1"
 serde_stacker = "0.1"
-regex = "1"
+regex = "~1.8"
```
Comment (jcnelson) [inline clarity/Cargo.toml:26]: Please avoid using tilde, and use a precise version.  We want all nodes everywhere to have the exact same code.

Comment (sergey-shandar) [reply clarity/Cargo.toml:26]: I though we are using [Cargo.lock](https://github.com/stacks-network/stacks-blockchain/blob/master/Cargo.lock) for this.

**Comment by jcnelson** [comment]
>  I still need clarification about the versioning strategy in the repo.

Just put the full version. Humans look at the cargo.toml file to determine
dependencies, not the cargo.lock.  The full version removes ambiguity.

**Comment by jcnelson** [comment]
> This PR doesn't touch the Cargo.toml at all right now

Ah I see that now. Will approve once I'm not AFK (give me an hour).

**Comment by sergey-shandar** [comment]
@kantai is it ok that the PR is open against the `master` branch? Do we sync changes from the master to the `develop` and `next` branches?

**Comment by kantai** [comment]
> @kantai is it ok that the PR is open against the master branch?

Yes -- for changes that don't touch any code (e.g., docs only, github workflows, etc.) it's okay to merge to master. This PR, because it just removes an unused import, is okay.

> Do we sync changes from the master to the develop and next branches?

Yes, changes to master should always be merged back into `develop` and `next`. There isn't an automated process for this in the github repository.

**Review by jcnelson — APPROVED** [review]

**Pull request merged**